File: Xamarin/GestioneMenuX/GestioneMenuX/GestioneMenuX.Android/MainActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using AndroidX.AppCompat.App;
using Java.IO;

namespace GestioneMenuX.Droid
{
    [Activity(Label = "@string/app_name", Icon = "@drawable/menu_logo", Theme = "@style/AppTheme", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        ListView listView;
        String[] vociListView = new string[] { "Context Uno", "Context Due", "Context Tre" };
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            listView = (ListView) FindViewById<ListView>(Resource.Id.listView);
            ArrayAdapter<string> adapter = new ArrayAdapter<string>(this, Android.Resource.Layout.SimpleListItem1, vociListView);
            listView.Adapter = adapter;
            RegisterForContextMenu(listView);

            Button btnPopUp = FindViewById<Button>(Resource.Id.popupButton);
            // setOnClickListener activates a listener
            btnPopUp.Click += (s, arg) =>
            {
                // creating the instance of popupmenu
                PopupMenu popupMenu = new PopupMenu(this, btnPopUp);
                // inflating the popup using xml file
                popupMenu.Inflate(Resource.Menu.popup_menu);
                // registering popup with OnMenuItemClickListener
                popupMenu.MenuItemClick += (s1, arg1) =>
                {
                    Toast.MakeText(this, arg1.Item.TitleFormatted, ToastLength.Long).Show();
                };
                // showing popup menu
                popupMenu.Show();
            };
        }
        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }

        // link the menu to the activity
        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            // this adds items to the action bar
            MenuInflater.Inflate(Resource.Menu.main_menu, menu);
            return base.OnCreateOptionsMenu(menu);
        }

        // the system calls the method when the user selects a menu item
        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            // identifies the item selected by the user
            int id = item.ItemId;
            switch (id)
            {
                case Resource.Id.voce_1:
                    Toast.MakeText(this, "Voce Uno", ToastLength.Long).Show();
                    break;
                case Resource.Id.voce_2:
                    Toast.MakeText(this, "Voce Due", ToastLength.Long).Show();
                    break;
                case Resource.Id.voce_3:
                    Toast.MakeText(this, "Voce Tre", ToastLength.Long).Show();
                    break;
            }
            return base.OnOptionsItemSelected(item);
        }

        // link the menu to the activity
        public override void OnCreateContextMenu(IContextMenu contextMenu, View v, IContextMenuContextMenuInfo menuInfo)
        {
            base.OnCreateContextMenu(contextMenu, v, menuInfo);
            // allows you to inflate the context menu from a menu resource
            MenuInflater.Inflate(Resource.Menu.option_menu, contextMenu);
            contextMenu.SetHeaderTitle("Seleziona l'Azione");
        }

        // the system calls the method when the user selects a menu item
        public override bool OnContextItemSelected(IMenuItem item)
        {
            // identifies the item selected by the user
            int id = item.ItemId;
            switch(id)
            {
                case Resource.Id.context_1:
                    Toast.MakeText(this, "Context Uno", ToastLength.Long).Show();
                    break;
                case Resource.Id.context_2:
                    Toast.MakeText(this, "Context Due", ToastLength.Long).Show();
                    break;
                case Resource.Id.context_3:
                    Toast.MakeText(this, "Context Tre", ToastLength.Long).Show();
                    break;
            }
            return base.OnContextItemSelected(item);
        }
    }
}

File: Android/GestioneMenu/app/src/main/java/com/example/gestionemenu/MainActivity.java
package com.example.gestionemenu;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.PopupMenu;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    ListView listView;
    String vociListView[] = {"Context Uno", "Context Due", "Context Tre", "Context Quattro", "Context Cinque"};
    Button btnPopUp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = (ListView)  findViewById(R.id.listView);
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, vociListView);
        listView.setAdapter(adapter);

        registerForContextMenu(listView);

        btnPopUp = (Button) findViewById(R.id.button);
        // setOnClickListener activates a listener
        btnPopUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // creating the instance of popupmenu
                PopupMenu popupMenu = new PopupMenu(MainActivity.this, btnPopUp);
                // inflating the popup using xml file
                popupMenu.getMenuInflater().inflate(R.menu.popup_menu, popupMenu.getMenu());
                // registering popup with OnMenuItemClickListener
                popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
                    @Override
                    public boolean onMenuItemClick(MenuItem menuItem) {
                        Toast.makeText(MainActivity.this, menuItem.getTitle(), Toast.LENGTH_LONG).show();
                        return false;
                    }
                });
                // showing popup menu
                popupMenu.show();
            }
        });
    }

    // link the menu to the activity
    public  boolean onCreateOptionsMenu(Menu menu) {
        // this adds items to the action bar
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }

    // link the menu to the activity
    public void onCreateContextMenu(ContextMenu contextMenu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(contextMenu, v, menuInfo);
        // allows you to inflate the context menu from a menu resource
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.option_menu, contextMenu);
        contextMenu.setHeaderTitle("Select The Action");
    }
    // the system calls the method when the user selects a menu item
    public boolean onOptionsItemSelected(MenuItem item) {
        // identifies the item selected by the user
        int id = item.getItemId();

        switch (id) {
            case R.id.voce_1:
                Toast.makeText(getApplicationContext(), "Main Uno", Toast.LENGTH_LONG).show();
                break;
            case R.id.voce_2:
                Toast.makeText(getApplicationContext(), "Main Due", Toast.LENGTH_LONG).show();
                break;
            case R.id.voce_3:
                Toast.makeText(getApplicationContext(), "Main Tre", Toast.LENGTH_LONG).show();
                break;
        }
        return false;
    }

    // the system calls the method when the user selects a menu item
    public boolean onContextItemSelected(MenuItem item) {
        // identifies the item selected by the user
        int id = item.getItemId();

        switch (id) {
            case R.id.context_1:
                Toast.makeText(getApplicationContext(), "Context Uno", Toast.LENGTH_LONG).show();
                break;
            case R.id.context_2:
                Toast.makeText(getApplicationContext(), "Context Due", Toast.LENGTH_LONG).show();
                break;
            case R.id.context_3:
                Toast.makeText(getApplicationContext(), "Context Tre", Toast.LENGTH_LONG).show();
                break;
        }
        return false;
    }
}